functions/helloworld/index.js

```diff
@@ -14,7 +14,76 @@
 'use strict';
 
 // [START helloworld]
-exports.helloworld = function (context, data) {
-  context.success('Hello World!');
+/**
+ * Cloud Function.
+ *
+ * @param {Object} context Cloud Function context.
+ * @param {Object} data Request data, provided by a trigger.
+ */
+exports.helloWorld = function helloWorld (context, data) {
+  console.log('My Cloud Function: ' + data.message);
+  context.success();
 };
 // [END helloworld]
+
+// [START helloGET]
+/**
+ * HTTP Cloud Function.
+ *
+ * @param {Object} req Cloud Function request context.
+ * @param {Object} res Cloud Function response context.
+ */
+exports.helloGET = function helloGET (req, res) {
+  res.send('Hello World!');
+};
+// [END helloGET]
+
+// [START helloHttp]
+/**
+ * HTTP Cloud Function.
+ *
+ * @param {Object} req Cloud Function request context.
+ * @param {Object} res Cloud Function response context.
+ */
+exports.helloHttp = function helloHttp (req, res) {
+  res.send('Hello ' + (req.body.name || 'World') + '!');
+};
+// [END helloHttp]
+
+// [START helloBackground]
+/**
+ * Background Cloud Function.
+ *
+ * @param {Object} context Cloud Function context.
+ * @param {Object} data Request data, provided by a trigger.
+ */
+exports.helloBackground = function helloBackground (context, data) {
```

Comment:
There are multiple background samples in the docs. Once the docs are published, you'll see somewhat of a mapping between sample folders here and pages in the docs.